spec/hashie/array_spec.rb

```diff
@@ -0,0 +1,17 @@
+require 'spec_helper'
+
+describe Array do
+  if RUBY_VERSION >= '2.3.0'
```

Comment:
Probably. I think it's not future proof. Especially worrisome in `Mash`. Feel free to make a separate PR though.